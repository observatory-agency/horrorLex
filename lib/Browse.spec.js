const Browse = require('./Browse');

describe('Browse', () => {
  let browse;
  const bookModelMock = {
    aggregate: jest.fn(() => ({
      toArray: jest.fn(() => [{
        _id: 1, documents: [{ title: 'book1' }],
      }]),
    })),
    find: jest.fn(() => ({
      toArray: jest.fn(() => [{ title: 'book1' }]),
    })),
  };

  beforeEach(() => {
    browse = new Browse(bookModelMock);
  });

  describe('constructor', () => {
    it('should set this.bookModel', () => {
      expect(browse.bookModel).toBe(bookModelMock);
    });
  });

  describe('createRegexStr', () => {
    describe('"a-zA-Z" character', () => {
      it('should return a formatted regex expressions', () => {
        const validChar = 'a';
        expect(Browse.createRegexStr(validChar)).toMatchSnapshot();
      });
    });
    describe('invalid characters', () => {
      it('should throw', () => {
        const invalidChar = '5';
        expect(() => {
          Browse.createRegexStr(invalidChar);
        }).toThrow('Invalid characters');
      });
    });
  });

  describe('byChar', () => {
    it('should return an object of formatted results', async () => {
      const regexChar = Browse.createRegexStr('a');
      const results = await browse.byChar(regexChar);
      expect(results).toMatchSnapshot();
    });
  });

  describe('relatedBooks', () => {

  });
});
