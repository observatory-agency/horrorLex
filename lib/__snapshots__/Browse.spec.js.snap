// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browse byChar should return an object of formatted results 1`] = `
Object {
  "results": Array [
    Object {
      "books": Array [
        Object {
          "title": "book1",
        },
      ],
      "title": 1,
    },
  ],
}
`;

exports[`Browse createRegexStr "a-zA-Z" character should return a formatted regex expressions 1`] = `"^[Aa]"`;
